src/controllers/colormapper.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <QColor>
+#include <QMap>
+#include <QObject>
+#include <QPair>
+#include <QVariant>
+
+#include "util/assert.h"
+
+class ColorMapper final {
+  public:
+    ColorMapper() = delete;
+    ColorMapper(const QMap<QRgb, QVariant> availableColors)
```

Comment:
- A `const` qualifier for parameters that are passed by-value is not recommended
- The constructor should be `explicit`